tensorflow/python/keras/backend.py

```diff
@@ -4023,12 +4023,16 @@ def dropout(x, level, noise_shape=None, seed=None):
   Returns:
       A tensor.
   """
-  retain_prob = 1. - level
   if seed is None:
     seed = np.random.randint(10e6)
   # the dummy 1. works around a TF bug
   # (float32_ref vs. float32 incompatibility)
-  return nn.dropout(x * 1., retain_prob, noise_shape, seed=seed)
+  try:
+    retain_prob = 1. - level if level is not None else None
+  except TypeError:
+    raise ValueError("You must provide a rate to dropout.")
```

Comment:
This should be structured as:

```
if level is None:
   raise ValueError('None values not supported. Received argument `level=None` in `keras.backend.dropout`.')
```

Reply:
@fchollet Updated the code as per your comments, please review.